zulip_bots/zulip_bots/lib_tests.py

```diff
@@ -138,3 +143,36 @@ def test_message(message, flags):
                                         config_file=None,
                                         bot_config_file=None,
                                         bot_name='testbot')
+
+    def test_upload_file(self):
+        client = FakeClient()
+        client.upload_file = MagicMock()
+
+        handler = ExternalBotHandler(
+            client=client,
+            root_dir=None,
+            bot_details=None,
+            bot_config_file=None
+        )
+
+        file = io.StringIO('text')
```

Comment:
Should this be `BytesIO`? (see my other query re text vs binary)

Reply:
@neiljp sure! changed to `BytesIO`